packages/nimble-components/src/rich-text/base/index.ts

```diff
@@ -41,24 +65,35 @@ export abstract class RichText extends FoundationElement {
      * which will be invoked in the RichTextMention base class from the client.
      */
     private updateMentionConfig(): void {
-        // TODO: Add a rich text validator to check if the `mentionElements` contains duplicate configuration element
-        // For example, having two `nimble-rich-text-mention-users` within the children of rich text viewer or editor is an invalid configuration
+        this.richTextValidator.validate(this.mentionElements);
         this.mentionConfig = [];
-        this.mentionElements.forEach(mention => {
-            if (mention.mentionInternals.validConfiguration) {
+        if (this.richTextValidator.isValid()) {
+            this.mentionElements.forEach(mention => {
                 const markdownParserMentionConfiguration = new MarkdownParserMentionConfiguration(
                     mention.mentionInternals
                 );
                 this.mentionConfig.push(markdownParserMentionConfiguration);
-            }
-        });
+            });
+        }
         this.updateView();
     }
 
     private childItemsChanged(): void {
         void this.updateMentionsFromChildItems();
     }
 
+    private mentionElementsChanged(
+        prev: string[] | undefined,
+        next: string[]
```

Comment:
As I responded in the other thread, I don't think this function is actually needed, but if for whatever reason it is, the types of `prev` and `next` should be `RichTextMention[]`, not `string[]`.